flang/lib/Lower/OpenMP.cpp

```diff
@@ -3032,7 +3037,16 @@ genTargetOp(Fortran::lower::AbstractConverter &converter,
         if (auto refType = baseOp.getType().dyn_cast<fir::ReferenceType>())
           eleType = refType.getElementType();
 
-        if (fir::isa_trivial(eleType) || fir::isa_char(eleType)) {
+        mlir::Value declOp = baseOp;
+        if (auto t = baseOp.getDefiningOp<hlfir::DeclareOp>())
```

Comment:
Did you mean to use `==` here ?
```suggestion
        if (auto t == baseOp.getDefiningOp<hlfir::DeclareOp>())
```

Reply:
No, it was a single =. It is similar logic to how addReductionDecl fills in the reductionVars.